src/test/java/org/example/tests/SortOptionsTest.java

```diff
@@ -0,0 +1,59 @@
+package org.example.tests;
+
+import org.example.enums.SortOption;
+import org.example.enums.UserCredentials;
+import org.example.pages.LoginPageBase;
+import org.example.pages.ProductPageBase;
+import org.example.utils.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class SortOptionsTest extends BaseTest {
+
+    @Test
+    public void testAllSortOptions() {
+        LoginPageBase loginPage = initPage(getDriver(), LoginPageBase.class);
+        ProductPageBase productPage = loginPage.login(UserCredentials.STANDARD_USER);
+
+        for (SortOption option : SortOption.values()) {
+            productPage.sortBy(option);
+
+            if (option == SortOption.PRICE_LOW_TO_HIGH || option == SortOption.PRICE_HIGH_TO_LOW) {
```

Comment:
Extract repeated logic (sorting and assertions)
here example:
public void testSortOptions() {
        ProductListPageBase productsPageBase = loginAsStandardUser(false);
        SoftAssert softAssert = new SoftAssert();
        SortOption[] sortOptions = {SortOption.A_TO_Z, SortOption.Z_TO_A, SortOption.LOW_TO_HIGH,
                SortOption.HIGH_TO_LOW};
        for (SortOption sortOption : sortOptions) {
            productsPageBase.checkCorrectSortingResult(softAssert, sortOption);
        }
        softAssert.assertAll();

    }